java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java

```diff
@@ -205,46 +220,67 @@ private PreparedStatement prepareDeleteStatement(SinkRow row) {
     public void write(Iterator<SinkRow> rows) {
         while (rows.hasNext()) {
             try (SinkRow row = rows.next()) {
-                PreparedStatement stmt;
                 if (row.getOp() == Data.Op.UPDATE_DELETE) {
                     updateFlag = true;
                     continue;
                 }
-
                 if (config.isUpsertSink()) {
-                    stmt = prepareForUpsert(row);
+                    prepareForUpsert(row);
                 } else {
-                    stmt = prepareForAppendOnly(row);
-                }
-
-                try {
-                    LOG.debug("Executing statement: {}", stmt);
-                    stmt.executeUpdate();
-                    stmt.clearParameters();
-                } catch (SQLException e) {
-                    throw Status.INTERNAL
-                            .withDescription(
-                                    String.format(ERROR_REPORT_TEMPLATE, stmt, e.getMessage()))
-                            .asRuntimeException();
+                    prepareForAppendOnly(row);
                 }
             } catch (Exception e) {
                 throw new RuntimeException(e);
             }
         }
+
+        // execute staging statement in batch after all rows are prepared,
+        // and we need to delete first to prevent accidentally deletion
+        var deleteStmt = stagingStatements.remove(OpType.DELETE);
```

Comment:
The current code seems too verbose with the three enum values.

We can simply has one individual stagingStatement for insert delete and update

Reply:
It seems that the three stmts only live within the scope of call to `write`? If so, I think we'd better store them as local variable instead of fields in the class.